### PR TITLE
Add protocol for simple access to the IPMI System Event Log

### PR DESCRIPTION
**Pull request by cfernald**
## Description

This pull request creates a DXE module and protocol to allow simple access to the IPMI System Event Log. This protocol is intended to be used by boot loaders, diagnostics tools, and other non-system components to allow less privileged access to the SEL. 

- [x] Impacts functionality?
- [ ] Impacts security?
- [ ] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Tested on local emulated machine.

## Integration Instructions

N/A

### CODE REVIEW

File: IpmiFeaturePkg/IpmiSel/IpmiSel.c [hunk 1]
```diff
@@ -0,0 +1,124 @@
+/** @file
+  The DXE implementation of the IPMI SEL Protocol.
+
+  Copyright (c) Microsoft Corporation
```
Comment (apop5) [inline IpmiFeaturePkg/IpmiSel/IpmiSel.c:4]:
```suggestion
  Copyright (c) Microsoft Corporation.
```

**Comment by apop5** [comment]
Looks good to me

File: IpmiFeaturePkg/Include/Library/IpmiSelLib.h [hunk 2]
```diff
@@ -113,6 +85,27 @@ SelAddOemEntry (
...
+  Adds an OEM timestamped event to the SEL.
+
+  @param[in,out]  RecordId         If provided, receives the record ID of the entry.
+  @param[in]      RecordType       The record type code. But be between 0xC0-0xDF.
+  @param[in]      ManufacturerId   The manufacturer ID for the event. Must be 3 bytes.
+  @param[in]      Data             Array of OEM defined event data.
+
+  @retval   EFI_SUCCESS             Event was successfully added to the SEL.
+  @retval   EFI_INVALID_PARAMETER   Invalid RecordType was given.
+  @retval   Other                   And error was returned by IpmiAddSelEntry.
```
Comment (makubacki) [inline IpmiFeaturePkg/Include/Library/IpmiSelLib.h:98]: I suggest dropping implementation details (like calling `IpmiAddSelEntry`) from the interface documentation.

File: IpmiFeaturePkg/IpmiSel/IpmiSel.inf [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  UefiBootServicesTableLib
+  UefiDriverEntryPoint
+  MemoryAllocationLib
+  UefiLib
+  DebugLib
+  BaseMemoryLib
+  IpmiSelLib
+
+[Protocols]
+  gIpmiSelProtocolGuid
```
Comment (makubacki) [inline IpmiFeaturePkg/IpmiSel/IpmiSel.inf:35]:
```suggestion
  gIpmiSelProtocolGuid  ## PRODUCES
```

File: IpmiFeaturePkg/Library/IpmiSelLib/IpmiSelLib.c [hunk 4]
```diff
@@ -213,25 +213,30 @@ SelAddSystemEntry (
...
   IN UINT8       RecordType,
+  IN UINT8       ManufacturerId[3],
   IN UINT8       Data[6]
   )
 
 {
   SEL_RECORD  Entry;
-  UINT8       *ManufacturerId;
+
+  if ((Data == NULL) || ((ManufacturerId == NULL))) {
```
Comment (makubacki) [inline IpmiFeaturePkg/Library/IpmiSelLib/IpmiSelLib.c:237]:
```suggestion
  if ((Data == NULL) || (ManufacturerId == NULL)) {
```

**Review by apop5 — APPROVED** [review]

**Review by makubacki — APPROVED** [review]

**Pull request merged**